### PR TITLE
feat: create a new react project example with a custom modal

### PR DESCRIPTION
**Pull request by omridan159**
Create a new react project example with how to create a custom QR code modal

### CODE REVIEW

File: docs/custom-modal.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Creating a Custom MetaMask Modal in React
+
+When integrating MetaMask into your React application, you might need to provide a custom user interface for certain interactions, such as displaying a modal when MetaMask is not installed. This documentation guides you through the process of creating and using a custom modal within the `MetaMaskProvider`.
```
Comment (deeeed) [inline docs/custom-modal.md:3]: maybe more generic, use web instead of react since we want that to work for any framework

**Review by deeeed — CHANGES_REQUESTED** [review]
I would jsut be more generic to describe web instead of react specific and say here is an implementation for react but same would apply for other framework such as vuejs

File: packages/examples/react-with-custom-modal/sdk-copy.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+reldir="$( dirname -- "$0"; )";
+cd "$reldir/../../..";
+SDK_WORKSPACE_DIR="$( pwd; )";
+COMM_LAYER_DIR="$SDK_WORKSPACE_DIR/packages/sdk-communication-layer"
+SDK_DIR="$SDK_WORKSPACE_DIR/packages/sdk"
+SDK_REACT_DIR="$SDK_WORKSPACE_DIR/packages/sdk-react"
+SDK_REACT_UI_DIR="$SDK_WORKSPACE_DIR/packages/sdk-react-ui"
+SDK_INSTALL_MODAL_WEB_DIR="$SDK_WORKSPACE_DIR/packages/sdk-install-modal-web"
+
+DAPP_DIR="$SDK_WORKSPACE_DIR/packages/examples/create-react-app"
```
Comment (christopherferreira9) [inline packages/examples/react-with-custom-modal/sdk-copy.sh:13]: Should this be?
DAPP_DIR="$SDK_WORKSPACE_DIR/packages/examples/react-with-custom-modal"

Comment (omridan159) [reply packages/examples/react-with-custom-modal/sdk-copy.sh]: Changed :)

**Review by christopherferreira9 — DISMISSED** [review]
LGTM

File: packages/examples/react-with-custom-modal/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Create React App SDK example
+
+This project was bootstrapped using [Create React App](https://github.com/facebook/create-react-app).
+
+Get started by running:
+
+```bash
+npm start
+```
+
```
Comment (abretonc7s) [inline packages/examples/react-with-custom-modal/README.md:10]: can we add a screenshot here to show the newly created modal?

**Review by abretonc7s — COMMENTED** [review]
just need a screenshot to highlight before after on the modal

**Review by abretonc7s — APPROVED** [review]

**Pull request merged**